triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -140,39 +140,39 @@ object Trigger extends StrictLogging {
     // Given an identifier to a high- or lowlevel trigger,
     // return an expression that will run the corresponding trigger
     // as a low-level trigger (by applying runTrigger) and the type of that expression.
-    def detectTriggerType(tyCon: TypeConName, tyArg: Type): Either[String, TypedExpr] = {
-      val triggerIds = TriggerIds(tyCon.packageId)
-      if (tyCon == triggerIds.damlTriggerLowLevel("Trigger")) {
-        logger.debug("Running low-level trigger")
-        val expr = EVal(triggerId)
-        val ty = TypeConApp(tyCon, ImmArray(tyArg))
-        Right(TypedExpr(expr, ty))
-      } else if (tyCon == triggerIds.damlTrigger("Trigger")) {
-        logger.debug("Running high-level trigger")
-
-        val runTrigger = EVal(triggerIds.damlTrigger("runTrigger"))
-        val expr = EApp(runTrigger, EVal(triggerId))
-
-        val triggerState = TTyCon(triggerIds.damlTriggerInternal("TriggerState"))
-        val stateTy = TApp(triggerState, tyArg)
-        val lowLevelTriggerTy = triggerIds.damlTriggerLowLevel("Trigger")
-        val ty = TypeConApp(lowLevelTriggerTy, ImmArray(stateTy))
-
-        Right(TypedExpr(expr, ty))
-      } else {
-        Left(s"Unexpected trigger type constructor $tyCon")
+    def detectTriggerType(triggerId: Identifier): Either[String, TypedExpr] = {
+      def error = Left(s"unknown trigger $triggerId")
```

Comment:
I think I don’t understand how this actually solves the issue. If I read this code correctly you’re now producing the same error for an identifier that doesn’t exist and one that has the wrong type. That seems pretty confusing. If you just want to shorten it maybe don’t print the actual type but print something like "identifier $triggerId is not a trigger".

Reply:
Is `s"identifier $triggerId is not a trigger"` very different from `s"unknown trigger $triggerId"` ?

Reply:
I think there are two user errors I can make:
1. I know the trigger I’m trying to reference and it really is a trigger but I typoed the name so there is nothing with that identifier.
2. I don’t really understand what I’m supposed to reference so I referenced the trigger rule instead of the trigger or something like that. The identifier exists but it has the wrong type.

I think providing enough info that users can distinguish which of the two they hit is useful.